Source/ISHRemote/Trisoft.ISHRemote/Objects/Public/IshSession.cs

```diff
@@ -80,54 +83,59 @@ public class IshSession : IDisposable
         /// Creates a session object holding contracts and proxies to the web services API. Takes care of username/password and 'Active Directory' authentication (NetworkCredential) to the Secure Token Service.
         /// </summary>
         /// <param name="webServicesBaseUrl">The url to the web service API. For example 'https://example.com/ISHWS/'</param>
+        /// <param name="wsTrustIssuerUrl">The url to the security token service wstrust endpoint. For example 'https://example.com/ISHSTS/issue/wstrust/mixed/username'</param>
+        /// <param name="wsTrustIssuerMexUrl">The binding for the wsTrustEndpoint url</param>
         /// <param name="ishUserName">InfoShare user name. For example 'Admin'</param>
         /// <param name="ishSecurePassword">Matching password as SecureString of the incoming user name. When null is provided, a NetworkCredential() is created instead.</param>
         /// <param name="timeout">Timeout to control Send/Receive timeouts of HttpClient when downloading content like connectionconfiguration.xml</param>
         /// <param name="timeoutIssue">Timeout to control Send/Receive timeouts of WCF when issuing a token</param>
         /// <param name="timeoutService">Timeout to control Send/Receive timeouts of WCF for InfoShareWS proxies</param>
         /// <param name="ignoreSslPolicyErrors">IgnoreSslPolicyErrors presence indicates that a custom callback will be assigned to ServicePointManager.ServerCertificateValidationCallback. Defaults false of course, as this is creates security holes! But very handy for Fiddler usage though.</param>
-        public IshSession(ILogger logger, string webServicesBaseUrl, string ishUserName, SecureString ishSecurePassword, TimeSpan timeout, TimeSpan timeoutIssue, TimeSpan timeoutService, bool ignoreSslPolicyErrors)
+        public IshSession(ILogger logger, string webServicesBaseUrl, string wsTrustIssuerUrl, string wsTrustIssuerMexUrl, string ishUserName, SecureString ishSecurePassword, TimeSpan timeout, TimeSpan timeoutIssue, TimeSpan timeoutService, bool ignoreSslPolicyErrors)
         {
             _logger = logger;
+            _explicitIssuer = true;
             _ignoreSslPolicyErrors = ignoreSslPolicyErrors;
             if (_ignoreSslPolicyErrors)
-            { 
+            {
                 CertificateValidationHelper.OverrideCertificateValidation();
             }
             ServicePointManagerHelper.RestoreCertificateValidation();
             // webServicesBaseUrl should have trailing slash, otherwise .NET throws unhandy "Reference to undeclared entity 'raquo'." error
-            _webServicesBaseUri = (webServicesBaseUrl.EndsWith("/")) ? new Uri(webServicesBaseUrl) : new Uri(webServicesBaseUrl+"/");
+            _webServicesBaseUri = (webServicesBaseUrl.EndsWith("/")) ? new Uri(webServicesBaseUrl) : new Uri(webServicesBaseUrl + "/");
+            _wsTrustIssuerUrl = new Uri(wsTrustIssuerUrl);
+            _wsTrustIssuerMexUrl = new Uri(wsTrustIssuerMexUrl);
```

Comment:
_wsTrustIssuerMexUrl should become _wsTrustIssuerMexUri (with an I instead of L)